code/game/objects/items/weapons/storage/fancy.dm

```diff
@@ -234,39 +234,45 @@
 /obj/item/storage/fancy/cigarettes/update_icon_state()
 	icon_state = "[initial(icon_state)][length(contents)]"
 
-/obj/item/storage/fancy/cigarettes/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
+/obj/item/storage/fancy/cigarettes/attack(mob/living/carbon/M, mob/living/user)
 	if(!ismob(M))
 		return
 
-	if(istype(M) && M == user && user.zone_selected == "mouth" && length(contents) > 0 && !user.wear_mask)
-		var/got_cig = 0
+	if(istype(M) && user.zone_selected == "mouth" && length(contents) > 0 && !M.wear_mask)
+		var/got_cig = FALSE
 		for(var/num=1, num <= length(contents), num++)
 			var/obj/item/I = contents[num]
 			if(istype(I, /obj/item/clothing/mask/cigarette))
 				var/obj/item/clothing/mask/cigarette/C = I
-				user.equip_to_slot_if_possible(C, SLOT_HUD_WEAR_MASK)
-				to_chat(user, "<span class='notice'>You take \a [C.name] out of the pack.</span>")
+				M.equip_to_slot_if_possible(C, SLOT_HUD_WEAR_MASK)
+				if(M != user)
+					user.visible_message(
+						"<span class='notice'>[user] takes \a [C.name] out of [src] and gives it to [M].</span>",
+						"<span class='notice'>You take \a [C.name] out of [src] and give it to [M].</span>"
+					)
+				else
+					to_chat(user, "<span class='notice'>You take \a [C.name] out of the pack.</span>")
 				update_icon()
-				got_cig = 1
+				got_cig = TRUE
 				break
```

Comment:
Why does this use a `for`-loop to grab the first cig from the pack ???
What were you smoking oldcoders????